ui/embeds/neetcode.py

```diff
@@ -0,0 +1,45 @@
+import discord
+from typing import TYPE_CHECKING
+from constants import Language
+from utils.problems import fetch_question_info, search_question
+from utils.neetcode import generate_neetcode_link
+from ui.embeds.problems import search_question, question_error_embed
+if TYPE_CHECKING:
+    # To prevent circular imports
+    from bot import DiscordBot
+
+# Neetcode Search by id
+async def search_neetcode_embed(bot: "DiscordBot", search_text: str, language: Language) -> discord.Embed:
+
+    try:
+        question_title, question_id = await search_question(bot, search_text)
+    except Exception as e:
+        return await question_error_embed()
```

Comment:
No need to catch exceptions, as ValueErrors and request related exceptions are caught within `search_question` and `bot.http_client.post_data` respectively.

Reply:
But make sure to check whether either `question_title` and `question_id` are empty, and if so, return nothing.

Reply:
question_error_embed isn't a coroutine, so no need to await.